solo_server/simple_setup.py

```diff
@@ -1,100 +1,333 @@
+         "large": "HuggingFaceTB/RobotGemma-3"
+    }
+}
+
+# -------------------------------
+# Utility: Print All-Models Table
+# -------------------------------
+def print_models_table():
+    from rich.table import Table
+    table = Table(title="Domain Model Mapping", border_style="blue")
+    table.add_column("Domain", style="cyan", no_wrap=True)
+    table.add_column("Small Model", style="magenta")
+    table.add_column("Medium Model", style="green")
+    table.add_column("Large Model", style="yellow")
+    for domain, mapping in domain_model_mapping.items():
+        table.add_row(domain, mapping["small"], mapping["medium"], mapping["large"])
+    console.print(table)
+
+# If the script is called with "all-models", show the table and exit.
+if len(sys.argv) > 1 and sys.argv[1] == "all-models":
+    print_models_table()
+    sys.exit(0)
+
+# -------------------------------
+# Interactive prompts
+# -------------------------------
+def get_user_input_domain():
+    """
+    Prompt the user to select a domain or enter a custom one.
+    """
+    domains = list(domain_model_mapping.keys()) + ["Custom"]
+    console.print(Panel("Select your domain from the following options:", border_style="blue"))
+    for idx, d in enumerate(domains, start=1):
+        console.print(f"[blue]{idx}.[/blue] {d}")
+    
+    choice = Prompt.ask("Enter the number corresponding to your domain", default=str(len(domains)))
+    try:
+        choice_num = int(choice)
+        if 1 <= choice_num < len(domains):
+            return domains[choice_num - 1]
+        else:
+            return Prompt.ask("Enter your custom domain")
+    except ValueError:
+        console.print("[red]Invalid input. Defaulting to 'Custom'.[/red]")
+        return Prompt.ask("Enter your custom domain")
+
+def get_user_input_role():
+    """
+    Prompt the user for their role.
+    """
+    return Prompt.ask("Enter your role (e.g., teacher, developer, data scientist)")
+
+def get_inference_engine_preference():
+    """
+    Ask the user to choose their default inference engine.
+    """
+    engines = ["ollama", "vllm", "sglang", "llamacpp"]
+    console.print(Panel("Select your preferred default inference engine:", border_style="blue"))
+    for idx, eng in enumerate(engines, start=1):
+        console.print(f"[blue]{idx}.[/blue] {eng}")
+    choice = Prompt.ask("Enter the number corresponding to your preferred inference engine", default="1")
+    try:
+        choice_num = int(choice)
+        if 1 <= choice_num <= len(engines):
+            return engines[choice_num - 1]
+        else:
+            console.print("[red]Invalid choice. Defaulting to ollama.[/red]")
+            return "ollama"
+    except ValueError:
+        console.print("[red]Invalid input. Defaulting to ollama.[/red]")
+        return "ollama"
+
+def get_file_type_preference():
+    """
+    Ask the user to choose their preferred file type.
+    """
+    types = ["onnx", "gguf"]
```

Comment:
FILE_TYPE_PREFERENCE = ["onnx","gguf"]

```suggestion
    types = ["onnx", "gguf"]
```